### DISCUSSION TITLE
Render final helmfile / helmfile package

### DISCUSSION THREAD

**Discussion by fivaldi** [helmfile/helmfile#534] [Ideas]
Current state: When using the `--debug` option, there's a possibility to get the final / rendered helmfile (by just parsing the output).

Idea: It would be great if there was an out-of-box subcommand, without the need of parsing output, e.g. `helmfile render`, which would provide the final helmfile output. If the `username` and `password` credentials were hidden/masked/censored by default in the repositories part/section, this would be safe by default. (Some `--unhide-credentials` option would do the other/opposite job, if somebody required this...)

The motivation is to provide a helmfile.yaml output for someone else (let's say customer), with configuration of releases, namespaces, chart versions etc., but without the actual credentials used. Let's assume the customer has his/her own credentials to downlaod the charts.

Advanced: `helmfile package` would do: `helmfile fetch`, `helmfile template`, `helmfile write-values` and `helmfile render` to a single (*.tgz) package, which would be highly reusable/convenient for any installation approach: k8s apply, manual helm commands, helmfile apply... :wink:

**Comment by yxxhero** [comment]
@fivaldi thanks for your idea. That sounds like good advice. PR is welcome.
